source/MaterialXRuntime/Codegen/RtSubGraphImpl.cpp

```diff
@@ -0,0 +1,55 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRuntime/Codegen/RtSubGraphImpl.h>
+#include <MaterialXRuntime/Tokens.h>
+
+#include <MaterialXRuntime/Private/PvtPath.h>
+#include <MaterialXRuntime/Private/PvtPrim.h>
+
+namespace MaterialX
+{
+
+DEFINE_TYPED_SCHEMA(RtSubGraphImpl, "nodeimpl:subgraphimpl");
```

Comment:
Just a comment really, do you think subgraphimpl is more specific than graphimpl? Not sure what makes it a "sub" graph.

Reply:
I used SubGraph instead of Compound here, which we had a long discussion about with Guillaume :)

I think it make sense to be more specific and include "Sub" in the name, since this will always be a sub-graph. Even in the case of a node in the top level using this implementation type this is still a sub-graph in the main top level node network.